index.js

```diff
@@ -306,6 +306,11 @@
     };
   };
 
+  var ValidDate = $.NullaryType(
+    'ValidDate',
+    function (x) { return $$typeEq('Date')(x) && !isNaN(x.getTime()) }
```

Comment:
Yeah, the reason I used `getTime` was because I read in the Mozilla [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/valueOf):

>This method is usually called internally by JavaScript and not explicitly in code.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/valueOf

But I'm happy to change it.